### ISSUE TITLE
key-spacing fails with `Value "[object Object]" no schemas match`

### ISSUE THREAD

**Issue by tomoguisuru** [eslint/eslint#8889]
* **ESLint Version:** v4.1.1
* **Node Version:** v6.10.2
* **npm Version:** v3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```javascript
module.exports = {
  //extends: 'eslint:recommended',
  rules: {
    "no-cond-assign": [2,"always"],
    "no-console": 1,
    "no-debugger": 2,
    "no-dupe-args": 2,
    "no-dupe-keys": 2,
    "no-duplicate-case": 2,
    "no-empty-character-class": 2,
    "no-empty": 2,
    "no-ex-assign": 2,
    "no-extra-parens": [1,"all"],
    "no-extra-semi": 2,
    "no-func-assign": 2,
    "no-invalid-regexp": 2,
    "no-negated-in-lhs": 1,
    "no-unreachable": 2,
    "use-isnan": 1,
    "valid-typeof": 2,
    "accessor-pairs": [2,{"setWithoutGet":true}],
    "block-scoped-var": 2,
    "curly": [2,"all"],
    "default-case": 1,
    "dot-location": [1,"property"],
    "eqeqeq": [2,"smart"],
    "no-alert": 2,
    "no-div-regex": 1,
    "no-else-return": 1,
    "no-floating-decimal": 2,
    "no-loop-func": 1,
    "no-catch-shadow": 2,
    "array-bracket-spacing": [2,"never"],
    "brace-style": [2,"1tbs",{"allowSingleLine":true}],
    "comma-spacing": [2,{"after":true}],
    "eol-last": 2,
    "key-spacing": [
            "error",
            {
                "singleLine": {
                    "beforeColon": false,
                    "afterColon" : true,
                    "align"      : "colon"
                },
                "multiLine": {
                    "beforeColon": false,
                    "afterColon" : true,
                }
            }
        ],
    "new-parens": 1,
    "no-inline-comments": 1,
    "no-multiple-empty-lines": [2,{"max":1}],
    "no-nested-ternary": 1,
    "no-trailing-spaces": 2,
    "object-curly-spacing": [2,"always",{"arraysInObjects":true,"objectsInObjects":true}],
    "quotes": [2,"single","avoid-escape"],
    "semi-spacing": [2,{"after":true}],
    "semi": [2,"always"],
    "sort-vars": 1,
    //"space-after-keywords": [2,"always"],
    "space-before-blocks": [2,"always"],
    "space-before-function-paren": [2,{"anonymous":"never","named":"never"}],
    "space-in-parens": [2,"never"],
    "space-infix-ops": [2,{"int32Hint":true}],
    //"space-return-throw-case": 2,
    "arrow-parens": [2,"as-needed"],
    "arrow-spacing": [2,{"before":true,"after":true}],
    "generator-star-spacing": [2,"after"],
    "no-const-assign": 2,
    "no-var": 1,
    "prefer-const": 1,
    "prefer-spread": 1,
    "require-yield": 1
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**

including the "align" option as detailed in the docs [key-spacing](http://eslint.org/docs/rules/key-spacing) results in an error, `Value "[object Object]" no schemas match`

Removing the "align" option nested under 'multiLine' or 'singleLine' resolves the error

```js
"key-spacing": [
            "error",
            {
                "singleLine": {
                    "beforeColon": false,
                    "afterColon" : true,
                    //"align"      : "colon"
                },
                "multiLine": {
                    "beforeColon": false,
                    "afterColon" : true,
                }
            }
        ],




```

**What did you expect to happen?**
it is expected that I can define the "align" option per the documentation

**What actually happened? Please include the actual, raw output from ESLint.**
an error, `Value "[object Object]" no schemas match` was displayed

**Comment by platinumazure** [comment]
Looking at the schema definition in the rule source, it looks like `align` can only be used in the `multiLine` options, or as a top-level option alongside `singleLine`/`multiLine`. Using `align` in a `singleLine` block does violate the schema, resulting in the unhelpful error message you are seeing. (N.B. We are hoping to switch JSON Schema validators soon, so that we can display better error messages.)

I'm not sure there is a use case for single-line alignment properties. Do you believe we should support that? If so, how would it work?

If we don't need to make a change, perhaps we could improve the documentation to emphasize that `align` only applies to (and makes sense in context with) multi-line settings. Do you have any concrete suggestions and/or would you be interested in writing a pull request?

**Comment by tomoguisuru** [comment]
probably no need to revise. I was just trying to enforce my own preferences through eslint
Example:
```js
{
  propA:          123,
  propB:          456,
  reallyLongName: 987,
  objA: {
    myPropA: 456,
    blah:    'blah',
  },
}
```

I'm probably trying to use the wrong rule anyway

**Comment by ilyavolodin** [comment]
Looks like the question has been answered. So I'm going to close this issue. Feel free to reopen if you disagree.
